src/Framework/Models/Factories/ModelFactory.php

```diff
@@ -85,16 +98,31 @@ public function create(array $attributes = [])
         return $this->count === 1 ? $instances[0] : $instances;
     }
 
+    /**
+     * @unreleased
+     */
+    public function createAndResolveTo($property)
+    {
+        return function() use ($property) {
+            return is_array($results = $this->create())
+                ? array_column($results, $property)
+                : $results->$property;
+        };
+    }
+
     /**
      * Creates an instance of the model from the attributes and definition.
      *
+     * @unreleased Add support for resolving callables.
      * @since 2.23.0
      *
      * @return M
      */
     protected function makeInstance(array $attributes)
     {
-        return new $this->model(array_merge($this->definition(), $attributes));
+        return new $this->model(array_map(function($attribute) {
+            return is_callable($attribute) ? $attribute() : $attribute;
+        }, array_merge($this->definition(), $attributes)));
```

Comment:
I wonder if the `is_callable()` is too generic? For example, model properties can be a class (say a VO), and if that class instance is invokable, then that would pass as `is_callable()` in which case it would be called here unintentionally.

I think it's safer to use `$attribute instanceof Closure` instead. I can't think of a time wherein a model property would be a closure, as that's not something persistable.

Reply:
Good point. I added a test to confirm and the test failed. Updating the resolver to check for a `Closure` type passed the test.

```php
    public function testDoesNotResolveInvokableClasses()
    {
        $factory = new class(MockModelWithInvokableProperty::class) extends ModelFactory {
            public function definition(): array {
                return [
                    'invokable' => new class extends MockInvokableClass {
                        public function __invoke() {
                            throw new \Exception('Invokable classes should not be resolved by factories.');
                        }
                    },
                ];
            }
        };

        $object = $factory->make();

        $this->assertInstanceOf(MockInvokableClass::class, $object->invokable);
    }
```

Reply:
Resolved, see [dd89630](https://github.com/impress-org/givewp/pull/7626/commits/dd896308a814d43cec9a6098301b95cc5c5a5656).